src/components/PatronBlock/PatronBlock.js

```diff
@@ -0,0 +1,158 @@
+import { useMemo } from 'react';
+import {
+  FormattedMessage,
+  useIntl,
+} from 'react-intl';
+import orderBy from 'lodash/orderBy';
+import PropTypes from 'prop-types';
+
+import {
+  Row,
+  Col,
+  Accordion,
+  MultiColumnList,
+  Headline,
+  Loading,
+  dayjs,
+  Icon,
+  Badge,
+} from '@folio/stripes/components';
+
+import {
+  useManualPatronBlocks,
+  useAutomatedPatronBlocks,
+} from '../../hooks';
+
+import css from './PatronBlock.css';
+
+const PATRON_BLOCKS_COLUMNS = {
+  type: 'type',
+  displayDescription: 'displayDescription',
+  blockedActions: 'blockedActions',
+};
+
+const columnMapping = {
+  [PATRON_BLOCKS_COLUMNS.type]: <FormattedMessage id="ui-reading-room.patronBlocks.columns.type" />,
+  [PATRON_BLOCKS_COLUMNS.displayDescription]: <FormattedMessage id="ui-reading-room.patronBlocks.columns.desc" />,
+  [PATRON_BLOCKS_COLUMNS.blockedActions]: <FormattedMessage id="ui-reading-room.patronBlocks.columns.blocked" />,
+};
+
+const columnWidths = {
+  [PATRON_BLOCKS_COLUMNS.type]: '10%',
+  [PATRON_BLOCKS_COLUMNS.displayDescription]: '45%',
+  [PATRON_BLOCKS_COLUMNS.blockedActions]: '45%',
+};
+
+const visibleColumns = [
+  PATRON_BLOCKS_COLUMNS.type,
+  PATRON_BLOCKS_COLUMNS.displayDescription,
+  PATRON_BLOCKS_COLUMNS.blockedActions,
+];
+
+const propTypes = {
+  userId: PropTypes.string.isRequired,
+};
+
+const PatronBlock = ({
+  userId,
+}) => {
+  const intl = useIntl();
+
+  const {
+    manualPatronBlocks,
+    isLoadingManualPatronBlocks,
+  } = useManualPatronBlocks({ userId });
+  const {
+    automatedPatronBlocks,
+    isLoadingAutomatedPatronBlocks,
+  } = useAutomatedPatronBlocks({ userId });
+
+  const patronBlocks = useMemo(() => {
+    // The filter keeps only patron blocks where the expiration date is either today or in the future
+    const blocks = [...manualPatronBlocks, ...automatedPatronBlocks].filter(patronBlock => {
+      return dayjs(patronBlock.expirationDate).endOf('day').isSameOrAfter(dayjs().endOf('day'));
+    });
+
+    return orderBy(blocks, ['metadata.createdDate'], ['desc']);
+  }, [manualPatronBlocks, automatedPatronBlocks]);
+
+  const isLoading = isLoadingManualPatronBlocks || isLoadingAutomatedPatronBlocks;
+  const totalRecords = patronBlocks.length;
+  const getCellClass = (defaultClass) => `${defaultClass} ${css.cellTopAlign}`;
+
+  const getPatronFormatter = () => {
+    return {
+      [PATRON_BLOCKS_COLUMNS.type]: f => {
```

Comment:
`f` - I suggest avoid variable name like a, b, c and etc. On my opinion we need use more precise names (that explain in detail what these variables do, store and etc.)